vach/src/loader/resource.rs

```diff
@@ -8,7 +8,7 @@ use crate::global::flags::Flags;
 #[derive(Debug)]
 pub struct Resource {
 	/// The processed data, stored as a vector of bytes `Vec<u8>`.
-	pub data: Vec<u8>,
+	pub data: Box<[u8]>,
```

Comment:
Why a box instead of a vec?

Reply:
`Box<[u8]>` is an immutable shrink_to_fitted buffer, plus the conversion is fundamentally free with this `Vec::<>u8::to_boxed_slice()`. Since in Rust immutablity is considered the default state, this is more in line than a `Vec<u8>`